### PR TITLE
Expo Template

### PR DESCRIPTION
**Pull request by takameyer**
Create a ts template for expo using Realm.

Also add a missing local .gitignore to all other templates.

This is using the beta version of Expo SDK 44 (which should be non-beta next week).
In order to include Expo Modules in the project, I had to downgrade to RN 0.64.2.

This closes # 4142

## ☑️ ToDos

* [x] Js Template
* [x] Android support
* [x] 📝 Changelog entry
* [ ] ~~Make symbolic links in the templates to hidden files that need to be active during development~~

### CODE REVIEW

File: packages/expo-template-js/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@realm/expo-template-js",
+  "version": "0.0.1-alpha.2",
```
Comment (kneth) [inline packages/expo-template-js/package.json:3]: Very defensive versioning :smile:

**Review by kneth — APPROVED** [review]
LGTM

Disclaimer: I have only sampled the files (picking with bias) and have't reviewed all files.

**Review by kraenhansen — APPROVED** [review]
Once the dot-files are symbolic links to the underscore files (as discussed in person) - this is 👍   to merge in my opinion.

**Comment by takameyer** [comment]
Unfortunately it seems git does not like symbolic links to `.gitignore`

**Pull request merged**